tasks/main.yml

```diff
@@ -0,0 +1,55 @@
+- name: Couchbase | Install Couchbase Repo
+  yum:
+    name: http://packages.couchbase.com/releases/couchbase-release/couchbase-release-1.0-0-x86_64.rpm
+    state: present
+
+- name: Couchbase | Install Couchbase
+  yum:
+    name: couchbase-server-4.5.0
+    state: present
+
+- name: Couchbase | Create Data Directory
+  file:
+    path: /data
+    state: directory
+    owner: couchbase
+    group: couchbase
+    mode: 0755
+
+- name: Couchbase | Enable Startup
+  service:
+    name: couchbase-server
+    state: started
+    enabled: yes
+
+- name: Couchbase | Check if Couchbase initialised
+  find:
+    paths: "{{ couchbase.datapath }}"
+    file_type: directory
+    patterns: "*"
+  register: dir_files
+
+- name: Couchbase | Check Buckets
+  shell: "/opt/couchbase/bin/couchbase-cli bucket-list -c localhost:8091 -u {{ couchbase.username }} -p {{ couchbase.password }}"
```

Comment:
Does this work? Or do you need to set the setting that allows a task to be failable (the exist code presumably is non-zero if couchbase is not configured?)

Reply:
This is just purely for checking if buckets are already there - the CLI isn't smart enough to skip existing bucket creation in the later steps and exits on a non zero causing Ansible tasks to fail out. I use the output from this (more specifically checking existance of bucket name) to verify if the bucket exists and create it if not

Reply:
This would be skipped if the cluster has been configured as per the when condition checking if the data path is empty (ie not configured)